source/administration/performance-alerting-guide.rst

```diff
@@ -0,0 +1,121 @@
+
+* CPU Utilization Rate
+* Memory Usage
+* Number of Goroutines
+* Number of API Errors per Second
+* Mean API Request Time
+
+To configure alerts 1) set an appropriate threshold and 2) enable notifications. Enabling notifications are the same for each chart, but setting the correct threshold can have some variances that are better handled on a per-chart basis.
+
+For each chart, click on the chart name and click “Edit”:
+
+.. image:: ../images/perf-1.png
+
+Then click on the “Alert” tab:
+
+.. image:: ../images/perf-2.png
+
+The alert threshold, which will be discussed in the sections below, is the last field under “Conditions” (the one set to 600 in the screenshot above).
+
+See the sections below for how to set the threshold for each individual chart. If you would like to add your own custom alert conditions, configure them here.
+
+.. image:: ../images/perf-3.png
+
+To enable the notifications for any alerts, click on the “Notification” tab on the left and select “Mattermost Alerts Channel” under “Send to”:
+
+.. image:: ../images/perf-4.png
+
+Fill in a message if you would like to add more context to the alert.
+
+By default, the alerts are configured to check the average of a chart over the last minute to see if that value is above a threshold. If it’s above the threshold, the alert will be triggered. Since it’s an average over the last minute, small spikes that go past the threshold won’t necessarily cause an alert. This helps prevent false positives that would result from natural spikes in usage. The alert state for each chart is evaluated every minute.
+
+The sections below describe each chart in more detail.
+
+CPU Utilization Rate
+---------------------
+
+CPU Utilization Rate is fairly straightforward. It tracks the CPU usage of the app servers as a percentage. The maximum percentage is based on the number of CPU cores or vCPUs your app server has. For example, if you have four CPU cores and your app server was at 100% utilization rate on all four cores, the graph would show 400% for that app server.
+
+It’s best to set the alert threshold based on your average CPU utilization and how many cores/vCPUs your app servers have. Take a look at the chart over the last seven days. You’ll want to set the threshold somewhere between your maximum CPU usage (cores * 100) and the CPU usage you see. The lower you set the threshold, the more alerts, and therefore the more false positives, you will get. Set the threshold too high and you may not receive an alert for an incident until it’s gotten worse. The same principle applies to all alerts, regardless of the chart.
+
+For example, on our daily build server that much of the team and community works out of, we have the threshold set to 15%:
+
+.. image:: ../images/perf-5.png
+
+This is well below our maximum CPU usage as we don’t have a high load. It’s also above our average usage at peak times, so we will get alerts if we start experiencing irregularly high CPU usage.
```

Comment:
`This value is below our maximum CPU usage and above our average usage at peak times. Therefore, we will get alerts if we begin experiencing unusually high CPU usage.`